### ISSUE TITLE
The getter 'isRequired' isn't defined for the class 'ParameterElement'.

### ISSUE THREAD

**Issue by Ludonope** [isar/isar#429]
**Steps to Reproduce**
I'm trying to start using isar, I also use `intl_generator` which only runs with `analyzer` `2.0.0` or `4.0.0` which forces me to use `isar_generator 3.0.0`.

I'm getting that crash and I don't have much idea why.
I'm not doing anything special (to my knowledge).

**Code sample**
```
[INFO] Generating build script...
[INFO] Generating build script completed, took 657ms

[INFO] Precompiling build script......
[WARNING] ../../../../flutter/.pub-cache/hosted/pub.dartlang.org/isar_generator-3.0.0-dev.1/lib/src/isar_analyzer.dart:89:18: Error: The getter 'isRequired' isn't defined for the class 'ParameterElement'.
 - 'ParameterElement' is from 'package:analyzer/dart/element/element.dart' ('../../../../flutter/.pub-cache/hosted/pub.dartlang.org/analyzer-4.0.0/lib/dart/element/element.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'isRequired'.
        (p) => p.isRequired && properties.none((e) => e.dartName == p.name));
                 ^^^^^^^^^^
[INFO] Precompiling build script... completed, took 6.5s

[SEVERE] Failed to precompile build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.

pub finished with exit code 78
```

**Version**
 - Platform: Mac
 - Flutter version: 3.0.2
 - Isar version: 3.0.0-dev.1
 - Isar Generator version: 3.0.0-dev.1

**Comment by GabrielCR99** [comment]
Facing the same issue here on Isar version 3.0.0-dev.1, on Windows 10. I'm only using isar required packages, being: isar, isar_flutter_libs and isar_generator.

**Comment by Ludonope** [comment]
As a note, the issue is not present in `isar 3.0.0-dev.0`.

**Comment by leisim** [comment]
There is a wrong dependency in the newest version. It requires `analyzer: ^4.1.0` but only specifies `analyzer: ^4.0.0`. You can use `dependency_overrides` until I can push a fixed release 🙂

**Comment by Ludonope** [comment]
Oh ok, only a small issue then. I would close it but I'll leave it to you in case you want to keep it open for tracking 😉

**Comment by leisim** [comment]
Fixed in `3.0.0-dev.2`
